esphome/components/tuya/tuya.cpp

```diff
@@ -398,42 +398,38 @@ void Tuya::send_local_time_() {
 }
 #endif
 
-void Tuya::set_datapoint_value(uint8_t datapoint_id, uint32_t value) {
-  ESP_LOGD(TAG, "Setting datapoint %u to %u", datapoint_id, value);
+void Tuya::set_raw_datapoint_value(uint8_t datapoint_id, const std::vector<uint8_t> &value) {
+  ESP_LOGD(TAG, "Setting datapoint %u to %s", datapoint_id, hexencode(value).c_str());
   optional<TuyaDatapoint> datapoint = this->get_datapoint_(datapoint_id);
   if (!datapoint.has_value()) {
-    ESP_LOGE(TAG, "Attempt to set unknown datapoint %u", datapoint_id);
+    ESP_LOGW(TAG, "Attempt to set unknown datapoint %u", datapoint_id);
```

Comment:
These should probably read as follows now to make it clear that whilst the datapoint is unknown we are actually setting it.
```suggestion
    ESP_LOGW(TAG, "Setting unknown datapoint %u", datapoint_id);
```